docs/en/query_language/agg_functions/reference.md

```diff
@@ -672,6 +672,112 @@ Optional parameters:
 - The default value for substituting in empty positions.
 - The length of the resulting array. This allows you to receive arrays of the same size for all the aggregate keys. When using this parameter, the default value must be specified.
 
+## groupArrayMovingSum {#agg_function-grouparraymovingsum}
+
+Calculates the moving sum of the input values.
+
+```
+groupArrayMovingSum(column_for_summing)
+groupArrayMovingSum(window_size)(column_for_summing)
+```
+
+The function can take the window size as a parameter. If it not specified, the function takes the window size equal to the column size.
+
+**Parameters**
+
+- `column_for_summing` — Column name. The column should have the numeric or [Decimal](../../data_types/decimal.md) data type.
+- `window_size` — Size of the sample window.
+
+**Returned values**
+
+- Array of the same size and type as the input data.
+
+**Example**
+
+The sample table:
+
+```text
+┌─a─┐
+│ 3 │
+│ 4 │
+│ 5 │
+│ 6 │
+└───┘
+```
+
+The queries:
+
+```sql
+SELECT groupArrayMovingSum(a) FROM b
+```
+```text
+┌─groupArrayMovingSum(a)─┐
+│ [3,7,12,18]            │
+└────────────────────────┘
+```
+```sql
+SELECT groupArrayMovingSum(2)(a) FROM b
+```
+```text
+┌─groupArrayMovingSum(2)(a)─┐
+│ [3,7,9,11]                │
+└───────────────────────────┘
+```
+
+## groupArrayMovingAvg {#agg_function-grouparraymovingavg}
+
+Calculates the moving average of the input values.
+
+```
+groupArrayMovingAvg(column_for_summing)
+groupArrayMovingAvg(window_size)(column_for_summing)
+```
+
+The function can take the window size as a parameter. If it not specified, the function takes the window size equal to the column size.
+
+**Parameters**
+
+- `column_for_summing` — Column name. The column should have the numeric or [Decimal](../../data_types/decimal.md) data type.
+- `window_size` — Size of the sample window.
+
+**Returned values**
+
+- Array of the same size and type as the input data.
+
+For integer numbers, the function truncates the decimal part of average values.
```

Comment:
so it rounds towards zero for integers? and does banking rounding for floats? probably it'd be easier to understand if explained in terms of rounding rules

btw this is way too inconsistent with normal `avg`...